openedx/features/genplus_features/genplus_learning/api/v1/serializers.py

```diff
@@ -156,7 +155,7 @@ def get_units(self, obj):
 
                 units_context[unit.pk] = {
                     'is_locked': is_locked,
-                    'progress': completion.progress if completion else 0,
+                    'progress': get_aggregated_progress(request._request, course_key),
```

Comment:
I think we should not use private variable(`_request`) here. What if we pass `request` as it is?

Reply:
It prompts exception, "The `request` argument must be an instance of `django.http.HttpRequest`, not `rest_framework.request.Request`" when calling view directly
<img width="665" alt="image" src="https://github.com/GenPlusLeaders/edx-platform/assets/19186089/148d4edf-132d-4889-810f-89097fc9a73e">